### DISCUSSION TITLE
No announcement when closing modal

### DISCUSSION THREAD

**Discussion by rhoadsstevens** [ManifoldScholar/manifold#3479] [Accessibility]
Priority level 2 (out of 3) from UW Accessible Technologies

WCAG mapping: 4.1.3 Status Messages

STEPS TO REPRODUCE:
1. Select a book (i.e. ""I My Recall"")
2. Turn on caret browsing (insert + F7)
3. Tab into the text
4. Arrow through the text to a comment. Press enter to open the comment
5. Press escape key or press the close button. Listen or watch for announcements using NVDA or other screen reader.

OBSERVED BEHAVIOR
No announcement when closing modal. When the modal closes, the screen reader starts announcing the annotation that you had originally selected to edit. Since it's not announcing the modal content anymore, it's technically possible to infer that you have returned to the book, but this requires a lot of additional cognitive processing that shouldn't be needed.

EXPECTED BEHAVIOR:
There needs to be announcements when there are dynamic changes.
When modal closes, there should be an announcement reflecting that, saying something such as "returning to book/text".

**Comment by tsmyre** [comment]
**CIC Response**
To the best of our knowledge, modal dialogs do not require announcements when they are closed. Moving focus back to a logical place is all that’s needed (and is something we need to fix). See https://www.tpgi.com/the-current-state-of-modal-dialog-accessibility/ and https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/.

**Michael Wilson**
Looking at the documentation we agree that it is not required, although it would be important information to have in some use cases. In the general use case of using the close button or escape from the modal, we agree that moving focus back to the logical place is sufficient, but consider the case where you delete the only comment related to an annotation, which automatically closes the modal with it. In this case, it may not be clear to SR users that deleting the comment would close the modal. Maybe instead of adding an announcement everytime the modal is closed, a solution to this specific use case would be to keep the modal open even if there are no comments so that users can actively decide to close the modal.

**CIC Response**
Thank you for clarifying! Separating things out like this so that the dialog isn’t closed as a side-effect of another user action makes sense.
